app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
 class ApplicationController < ActionController::Base
+  helper_method :user_logged_in?
+  def remember(user)
+    user.token_encrypt
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.token
+  end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      session[:user_id] = user.id
+      @current_user = user
+    end
+  end
```

Comment:
Too long. Rewrite it by keeping in my that when a user is connected only the remember token is set up in cookies.permanent.